src/sinks/humio.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    sinks::splunk_hec::{Encoding, HecSinkConfig},
+    sinks::util::{BatchBytesConfig, TowerRequestConfig},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use serde::{Deserialize, Serialize};
+
+const HOST: &str = "https://cloud.humio.com";
```

Comment:
Is there a reason you made this a const here instead of the default for the `host` field?

Reply:
I'm learning Rust, so I was wondering about this too...

Reply:
We generally use consts or lazy_static for things like well known endpoints like this cloud one.

In general, we use `Option<...>` with our `serde` configs to denote that this field is optional but the default for `Option<...>` is always `None`. This is fine, because we are forced to handle this case within config building. We generally, prefer to set defaults in the `SinkConfig::build` phase via `unwrap_or`'s. This allows the most flexibility.

But as you said, its possible to use defaults and `serde` to do some of this magic. Aka `#[serde(default)]`. The reason I generally prefer not to do this is because it is actually much more verbose for not much benefit. Because you either need to newtype and implement default or you use `#[serde(default_path = "my_default_fn"]` which then requires you to define a fn to statically return a type.

In reality these fn just look kind of noisy and not very useful where as unwrapping them in a well known spot is easy to find and isn't noisy.